src/neo/Ledger/Blockchain.cs

```diff
@@ -429,7 +429,9 @@ private void OnParallelVerified(ParallelVerified parallelVerified)
             RelayResultReason reason = parallelVerified.VerifyResult;
             if (reason == RelayResultReason.Succeed)
             {
-                if (!MemPool.CanTransactionFitInPool(parallelVerified.Transaction))
+                if (View.ContainsTransaction(parallelVerified.Transaction.Hash))
```

Comment:
It has been check in L403.

Reply:
But it cannot guarantee that transactions are not duplicated after asynchronous verification, because during the asynchronous verification process, there may be blocks containing the transaction that are persisted to storage.

Reply:
I think that he mentioned that it checks but meanwhile it since it not thread safe a Block can be Persisted during the Asynchronous loop. But I am not sure if I understood correctly.

Reply:
aehuahuea, we send almost together, @eryeer.